src/Scorecard.js

```diff
@@ -0,0 +1,86 @@
+var Scorecard = function() {
+  this.rolls = [];
+  this.frameTotals = [];
+};
+
+Scorecard.prototype.roll = function(roll1, roll2, roll3) {
+  if (typeof roll3 === 'undefined') {roll3 = 0;}
+  if (roll1 > 10 || roll2 > 10 || roll3 > 10) {
+    throw new Error("can't score more than 10!");
+  }
+  if (((roll1 + roll2) > 10) && (this.rolls.length != 9)) {
+    throw new Error("You cannot score more than 10 from two rolls!");
+  }
+  if (this.rolls.length === 9) {
+    if (((roll1 || roll2) < 10) && ((roll1 + roll2) < 10)) {
+      if (roll3 > 0) {
+        throw new Error("Cannot use 3rd roll as you did not have a strike");
+      }
+    } else {
+      return this.rolls.push([roll1, roll2, roll3]);
+    }
+  }
+  if (this.rolls.length === 10) {
+    throw new Error("You cannot play more than 10 frames!");
+  }
+  this.rolls.push([roll1, roll2]);
+};
```

Comment:
Too many conditionals. Use an object to describe the rules, or an index to hold the current strike / spare status.